### PR TITLE
2178 apiv3 labels

### PR DESCRIPTION
**Pull request by aviveiros11**
#### Any background context you want to provide?
The effort to move to API v3 - specifically labels
#### What's this PR do?
translates API v2 into v3 with auto schema helper 
#### How should this be manually tested?
Use the `Try It` functionality rendered by the API page on SEED for v3
#### What are the relevant tickets?
#2178

### CODE REVIEW

File: seed/static/seed/js/services/label_service.js [hunk 1]
```diff
@@ -69,15 +69,19 @@ angular.module('BE.seed.service.label', [])
...
+            params: params
+          }).then(map_labels);
         }
-        return $http.post('/api/v2/labels/filter/', body, {
-          params: params
-        }).then(map_labels);
-      }
 
+      }
+
```
[the remark this answers is not in the document]
Comment (aviveiros11) [reply seed/static/seed/js/services/label_service.js:84]: Yes, sorry the motivation behind this change was to confirm its functionality when actually implemented on the front end.

[the remark this answers is not in the document]
Comment (adrian-lara) [reply seed/static/seed/js/services/label_service.js:84]: Oh good point - maybe keep it for now until this PR is ready to be submitted?

File: seed/views/v3/labels.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+)
+from seed.utils.api import drf_api_endpoint
+from seed.utils.api_schema import swagger_auto_schema_org_query_param
+from seed.utils.labels import _get_labels
+from seed.utils.viewsets import SEEDOrgNoPatchOrOrgCreateModelViewSet
+
+ErrorState = namedtuple('ErrorState', ['status_code', 'message'])
+
+
+class LabelViewSet(DecoratorMixin(drf_api_endpoint), SEEDOrgNoPatchOrOrgCreateModelViewSet):
```
[the remark this answers is not in the document]
Comment (adrian-lara) [reply seed/views/v3/labels.py:74]: Nice - Let's now update the descriptions for these to mention that the org id is optional and defaults to the user's currently active organization, and I think these DRF-generated endpoints will be good to go.

File: seed/views/v3/labels.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    decorator=swagger_auto_schema(
+        manual_parameters=[AutoSchemaHelper.query_org_id_field(required=False)]
+    ),
+)
+@method_decorator(
+    name='list',
+    decorator=swagger_auto_schema(
+        manual_parameters=[AutoSchemaHelper.query_org_id_field(required=False)]
+    ),
+)
```
Comment (adrian-lara) [inline seed/views/v3/labels.py:39]: There's a chance we still want to keep the custom logic from the previous iteration of this list endpoint. Let's chat about this one tomorrow. We should probably review what that custom logic was allowing us to do. (See comment about the `_get_labels` method in this file.)

File: seed/utils/labels.py [hunk 4]
```diff
@@ -17,8 +17,37 @@
 )
 
 
-def _get_labels(request, qs, super_organization, inv_type):
-    inventory = InventoryFilterBackendWithInvType().filter_queryset_with_inv(
+def filter_labels_for_inv_type(request, inv_type):
```
Comment (adrian-lara) [inline seed/utils/labels.py:20]: See comment about defaulting inv_type to None

File: seed/views/v3/labels.py [hunk 5]
```diff
@@ -61,20 +107,8 @@ def get_queryset(self):
 
     def get_serializer(self, *args, **kwargs):
         kwargs['super_organization'] = self.get_organization(self.request)
-        inventory = InventoryFilterBackendWithInvType().filter_queryset_with_inv(
+        inventory = filter_labels_for_inv_type(
             request=self.request, inv_type=None
```
Comment (adrian-lara) [inline seed/views/v3/labels.py:108]: What do you think about just defaulting inv_type to None so you don't have to provide it here?

File: seed/utils/labels.py [hunk 6]
```diff
@@ -17,8 +17,37 @@
...
+    """
+    Method used to filter labels by inventory type and return is_applied inventory id's
+    Method was initially built as a class defined function above to handle more parameters.
+    """
+    params = request.query_params.dict()
+    # Since this is being passed in as a query string, the object ends up
+    # coming through as a string.
+    params['filter_params'] = json.loads(params.get('filter_params', '{}'))
+    inventory_type = inv_type
```
Comment (adrian-lara) [inline seed/utils/labels.py:29]: Let's remove this extra variable assignment and use inv_type directly below. We could also just rename the argument to be "inventory_type", and delete this line.

File: seed/views/v3/taxlots.py [hunk 7]
```diff
@@ -10,31 +11,36 @@
...
+            description='An array of taxlot ids to be returned'
+        )
+    )
     @action(detail=False, methods=['POST'])
     def labels(self, request):
         """
         Returns a list of all labels where the is_applied field
         in the response pertains to the labels applied to taxlot_view
         """
         inv_type = 'taxlot_view'
```
Comment (adrian-lara) [inline seed/views/v3/taxlots.py:40]: What do you think about just putting 'taxlot_view' directly as an arg in line 46 and removing this variable declaration?

File: seed/views/v3/properties.py [hunk 8]
```diff
@@ -12,31 +13,36 @@
...
+            description='An array of property ids to be returned'
+        )
+    )
     @action(detail=False, methods=['POST'])
     def labels(self, request):
         """
         Returns a list of all labels where the is_applied field
         in the response pertains to the labels applied to property_view
         """
         inv_type = 'property_view'
```
Comment (adrian-lara) [inline seed/views/v3/properties.py:42]: What do you think about just putting 'property_view' directly as an arg in line 46 and removing this variable declaration?

File: seed/views/v3/labels.py [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+@method_decorator(
+    name='list',
+    decorator=swagger_auto_schema(
+        manual_parameters=[AutoSchemaHelper.query_org_id_field(required=False)]
+    ),
+)
+@method_decorator(
+    name='create',
+    decorator=swagger_auto_schema(
+        manual_parameters=[AutoSchemaHelper.query_org_id_field(required=False, description="put in org id description")],
```
Comment (adrian-lara) [inline seed/views/v3/labels.py:43]: Reminder for this change request on this and other `query_org_id_field`s. The description should mention that  the param is optional and will default to the user's currently selected organization.

File: charts/backend/templates/db-postgres-deployment.yaml [hunk 10]
```diff
@@ -32,7 +32,7 @@ spec:
         image: seedplatform/postgres-seed:11.2
         name: db-postrgres
 #        command: ["tail"]
-#        args: ['-f', '/dev/null']
+#        args: ['-f', '/dev/n
```
Comment (adrian-lara) [inline charts/backend/templates/db-postgres-deployment.yaml:35]: Is this an inadvertent change?

File: seed/views/v3/properties.py [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+ErrorState = namedtuple('ErrorState', ['status_code', 'message'])
+
+
+class PropertyViewSet(viewsets.ViewSet, OrgMixin):
+    renderer_classes = (JSONRenderer,)
+    parser_classes = (JSONParser,)
+    _organization = None
+
+    @swagger_auto_schema(
+        manual_parameters=[AutoSchemaHelper.query_org_id_field(required=False)],
```
Comment (adrian-lara) [inline seed/views/v3/properties.py:28]: Org ID seems to be required in this case.

File: seed/views/v3/taxlots.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+ErrorState = namedtuple('ErrorState', ['status_code', 'message'])
+
+
+class TaxlotViewSet(viewsets.ViewSet, OrgMixin):
+    renderer_classes = (JSONRenderer,)
+    parser_classes = (JSONParser,)
+    _organization = None
+
+    @swagger_auto_schema(
+        manual_parameters=[AutoSchemaHelper.query_org_id_field(required=False)],
```
Comment (adrian-lara) [inline seed/views/v3/taxlots.py:26]: Org ID seems to be required in this case.

File: seed/views/v3/labels.py [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+from seed.utils.labels import filter_labels_for_inv_type
+from seed.utils.viewsets import SEEDOrgNoPatchOrOrgCreateModelViewSet
+
+
+@method_decorator(
+    name='retrieve',
+    decorator=swagger_auto_schema(
+        manual_parameters=[AutoSchemaHelper.query_org_id_field(required=False, description="Optional org id which "
+                                                                                           "overrides the users "
+                                                                                           "(default) current org id")]
```
Comment (adrian-lara) [inline seed/views/v3/labels.py:33]: Can we change the format of these? It makes it visually more difficult to pick up on what's going on to see the description broken up like this. (These method_decorators are already visually hard to track - we don't want to add to make it worse.)

Maybe have `required` and `description` be on their own lines?
```
        manual_parameters=[AutoSchemaHelper.query_org_id_field(
            required=False,
            description="Optional org id which overrides the users (default) current org id"
        )]
```

**Pull request merged**